### ISSUE TITLE
Missing context information after "IDFAQuery" event from IdfaPlugin

### ISSUE THREAD

**Issue by i8ramin** [segmentio/analytics-react-native#552]
It appears that after the `IDFAQuery` event is triggered (via the `IdfaPlugin` plugin), all subsequent events are missing the device information (which was there before and during the `IDFAQuery` event)

- `analytics-react-native` version: `2.2.0`
- Integrations versions (if used):
-- `@segment/analytics-react-native-plugin-idfa` @ `0.2.1`
-- `@segment/analytics-react-native-plugin-facebook-app-events` @ `0.2.1`
-- `@segment/analytics-react-native-plugin-firebase` @ `0.2.1`
- React Native version: `0.64.3`
- iOS or Android or both? only tested on `iOS`

**Steps to reproduce**

I am initializing the Segment client as follows:

```ts
import { createClient } from '@segment/analytics-react-native'
import { FacebookAppEventsPlugin } from '@segment/analytics-react-native-plugin-facebook-app-events'
import { FirebasePlugin } from '@segment/analytics-react-native-plugin-firebase'
import { IdfaPlugin } from '@segment/analytics-react-native-plugin-idfa'

const segmentClient = createClient({
  writeKey: process.env.ANALYTICS_WRITE_KEY as string,
  trackAppLifecycleEvents: true,
  trackDeepLinks: true,
  collectDeviceId: true,
  debug: true,
})

segmentClient.add({ plugin: new IdfaPlugin() })
segmentClient.add({ plugin: new FacebookAppEventsPlugin() })
segmentClient.add({ plugin: new FirebasePlugin() })

export default segmentClient
```

When I load up my app, I get the following events triggered and sent to Segment:

- TRACK `Application Installed` (has device info)
```json
    "device": {
      "adTrackingEnabled": false,
      "advertisingId": "955C8056-AC93-4FE6-8B40-XXXXXXXXXXX",
      "id": "955C8056-AC93-4FE6-8B40-XXXXXXXXXXX",
      "manufacturer": "Apple",
      "model": "x86_64",
      "name": "iPhone",
      "trackingStatus": "notDetermined",
      "type": "ios"
    },
```
- TRACK `IDFAQuery` (has device info)
```json
    "device": {
      "adTrackingEnabled": false,
      "advertisingId": "955C8056-AC93-4FE6-8B40-XXXXXXXXXXX",
      "id": "955C8056-AC93-4FE6-8B40-XXXXXXXXXXX",
      "manufacturer": "Apple",
      "model": "x86_64",
      "name": "iPhone",
      "trackingStatus": "notDetermined",
      "type": "ios"
    },
```
- SCREEN `Login` (_missing_ device info)
```json
    "device": {
      "adTrackingEnabled": false,
      "advertisingId": "955C8056-AC93-4FE6-8B40-XXXXXXXXXXX",
      "trackingStatus": "denied"
    },
```
- TRACK `Tapped Log In` (_missing_ device info)
```json
    "device": {
      "adTrackingEnabled": false,
      "advertisingId": "955C8056-AC93-4FE6-8B40-XXXXXXXXXXX",
      "trackingStatus": "denied"
    },
```

**Expected behavior**

I would expect that all of the track/screen events have device info.

**Actual behavior**

Device info is missing from context after the IDFAQuery event is triggered

**Comment by i8ramin** [comment]
Seeing very strange behavior. I re-installed app (doing this on simulator with `expo run:ios` command btw), and even though I am opting into ad tracking, the subsequent tracking calls all have `adTrackingEnabled: false`. 

```json
{
  "anonymousId": "FPiU-wNXtdt1wPLTPj3F7",
  "context": {
    "app": {
      "build": "20",
      "name": "Carrot",
      "namespace": "com.addtocarrot.com.carrot",
      "version": "0.0.26"
    },
    "device": {
      "adTrackingEnabled": false, <-- this should be `true` no?
      "advertisingId": "3789FEF7-D590-4840-A4BE-XXXXXXXXXXX",
      "id": "3789FEF7-D590-4840-A4BE-XXXXXXXXXXX",
      "manufacturer": "Apple",
      "model": "x86_64",
      "name": "iPhone",
      "trackingStatus": "notDetermined",
      "type": "ios"
    },
  },
  "event": "IDFAQuery",
  "properties": {
    "result": "authorized" <--- AUTHORIZED
  },
  "type": "track",
}
```

Following the above event, the next few ones have device information that looks like this:
```json
    "device": {
      "adTrackingEnabled": true,
      "advertisingId": "00000000-0000-0000-0000-000000000000",
      "trackingStatus": "authorized"
    },
```

Where the `adTrackingEnabled` value is correctly set, but device information is missing again, and also the `advertisingId` value is set to all 000's

Finally, when I refresh my app (via expo cli), the events have the following device info. Correct `adTrackingEnabled`, proper device info but 000's as `advertisingId`
```json
    "device": {
      "adTrackingEnabled": true,
      "advertisingId": "00000000-0000-0000-0000-000000000000",
      "id": "3789FEF7-D590-4840-A4BE-XXXXXXXX",
      "manufacturer": "Apple",
      "model": "x86_64",
      "name": "iPhone",
      "trackingStatus": "authorized",
      "type": "ios"
    },
```

I am utterly confused at this point.

**Comment by oscb** [comment]
@i8ramin this looks like a bug that was caused by the IDFA plugins overwriting all the context. We released v2.2.1 with a fix for deepmerging the device context so that this doesn't happen: https://github.com/segmentio/analytics-react-native/releases/tag/v2.2.1 can you give it a try and see if it fixes your issue?

I have the feeling that `adTrackingEnabled` not showing up as enabled right away could be a different bug though. I'll investigate more into it. But v2.2.1. should fix the bigger issue for now.

**Comment by oscb** [comment]
This should be fixed in v2.5.0

**Comment by pjuchniewicz** [comment]
I am encountering the same issue with  even the latest versions.

```
segmentClient = createClient({
    writeKey: Config.SEGMENT_WRITE_KEY,
    trackAppLifecycleEvents: true,
    trackDeepLinks: true,
  });
  segmentClient.add({plugin: new IdfaPlugin()});

// commented out for test 4
  const plugin = new AppsflyerPlugin({
      devKey: Config.APPSFLYER_SDK_KEY,
      isDebug: Config.IS_DEBUG_ENV,
      appId: Config.IOS_APP_ID,
      onInstallConversionDataListener: true, //Optional
      onDeepLinkListener: true, //Optional
      timeToWaitForATTUserAuthorization: 10, //for iOS 14.5
    });
    segmentClient.add({plugin});
```

react-native: 0.66.4

Test 1:
- @segment/analytics-react-native: ^2.3.2
- @segment/analytics-react-native-plugin-appsflyer: ^0.3.3
- @segment/analytics-react-native-plugin-idfa: ^0.4.1

Test 2:
- @segment/analytics-react-native: ^2.5.1
- @segment/analytics-react-native-plugin-appsflyer: ^0.3.3
- @segment/analytics-react-native-plugin-idfa: ^0.4.1

Test 3:
- @segment/analytics-react-native: ^2.5.1
- @segment/analytics-react-native-plugin-appsflyer: ^0.4.0
- @segment/analytics-react-native-plugin-idfa: ^0.4.1

Test 3:
- @segment/analytics-react-native: ^2.5.1
- @segment/analytics-react-native-plugin-idfa: ^0.4.1
Removed the appsflyer plugin so only segment and the idfa plugin


Same results for all tests:

IDFAQuery:
- status is not determined
- tracking enabled is false
- has advertising id

```
"device": {
      "adTrackingEnabled": false,
      "advertisingId": "41D8A982-XXXX",
      "id": "41D8A982-XXXX",
      "manufacturer": "Apple",
      "model": "x86_64",
      "name": "iPhone",
      "trackingStatus": "notDetermined",
      "type": "ios"
    },
```

Subsequent events:
- status is authorized
- tracking enabled is true
- advertising id is all zeros
```
"device": {
      "adTrackingEnabled": true,
      "advertisingId": "00000000-0000-0000-0000-000000000000",
      "id": "41D8A982-XXXX",
      "manufacturer": "Apple",
      "model": "x86_64",
      "name": "iPhone",
      "trackingStatus": "authorized",
      "type": "ios"
    },
```
